qiskit/algorithms/gradients/base_sampler_gradient.py

```diff
@@ -54,7 +54,7 @@ def __init__(self, sampler: BaseSampler, options: Options | None = None):
         self._default_options = Options()
         if options is not None:
             self._default_options.update_options(**options)
-        self._gradient_circuit_cache: dict[QuantumCircuit, GradientCircuit] = {}
+        self._gradient_circuit_cache: dict[QuantumCircuit | tuple, GradientCircuit] = {}
```

Comment:
similar comment to the QGT, shouldn't this typehint match the cache of base_estimator_gradient.py?